### DISCUSSION TITLE
Question: Is it possible to load Vue component locally in 1 liquid file?

### DISCUSSION THREAD

**Discussion by albertpratomo** [uicrooks/shopify-theme-lab#132] [Q&A] [answered]
I don't want to register all of my Vue components globally. Is it possible to register locally in 1 liquid file?

**Comment by sergejcodes** [comment, marked as answer]
Wouldn't recommend it, this adds unnecessary complexity.

If you want to do it anyway, you should probably try the modular approach described in the docs [here](https://themelab.uicrooks.com/guide/modularity.html#modular-approach). And register the components manually, something like this (in main.js):

```js
import './my-component.vue'

const createVueAppForMyComponent = () => {
  const app = createApp({})
  app.component('my-component', component)
  return app
}

createVueAppForMyComponent().mount('#vue-for-my-component')
```

```liquid
<div id="vue-for-my-component"></div>
```
